### ISSUE TITLE
pylint-1.5.4 false deprecation warning when using six and unittest

### ISSUE THREAD

**Issue by bcl** [PyCQA/pylint#806]
To reproduce this create a directory and copy this into __init__.py

```python
import six
import unittest
from unittest import TestCase

# deprecation compatibility b/w python 2 and 3
if six.PY2:
    TestCase.assertRaisesRegex = TestCase.assertRaisesRegexp
    TestCase.assertRegex = TestCase.assertRegexpMatches

class RegexTestCase(unittest.TestCase):
    def first(self):
        with self.assertRaisesRegex(RuntimeError, "runtime error"):
            raise RuntimeError
```

With python 3.5 pylint complains that assertRaisesRegex is deprecated, which it isn't, it is the new name for the deprecated assertRaisesRegexp

**Comment by PCManticore** [comment]
This happens because when inferring, we're also taking the if branch and assertRaisesRegex is inferred as assertRaisesRegexp. Without having some flow control filtering, we don't have too many options for this. 

But as a solution which works on both versions without requiring a fix in Pylint is to use six.assertRaisesRegex which does the same job and it's compatible with both versions.
